ui/contexts/AuthContext.tsx

```diff
@@ -80,21 +92,43 @@ export default function AuthContextProvider({ children }) {
       .finally(() => setLoading(false));
   }, []);
 
+  const logOut = React.useCallback(() => {
+    setLoading(true);
+    fetch(LOG_OUT, {
+      method: "POST",
+    })
+      .then((response) => {
+        if (response.status !== 200) {
+          setError(response);
+          return;
+        }
+        history.push("/sign_in");
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
   React.useEffect(() => {
     getUserInfo();
     return history.listen(getUserInfo);
   }, [getUserInfo, history]);
```

Comment:
Maybe wrap this in a featureflag too?


```suggestion
  React.useEffect(() => {
    if (!authFlag) {
      return null;
    }
    getUserInfo();
    return history.listen(getUserInfo);
  }, [getUserInfo, history]);
```